src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/AbstractRazorDelegatingEndpoint.cs

```diff
@@ -46,6 +47,22 @@ internal abstract class AbstractRazorDelegatingEndpoint<TRequest, TResponse> : I
     /// </summary>
     protected virtual bool PreferCSharpOverHtmlIfPossible { get; } = false;
 
+    /// <summary>
+    /// When <see langword="true"/>, we'll adjust the position of the caret to the name portion of an attribute if it is in the
+    /// attribute, but not the name. eg. if we get @bi$$nd-Value we'll act as though we got @bind-$$Value
+    /// </summary>
+    protected virtual bool TreatAnyAttributePositionAsAttributeName { get; } = false;
+
+    /// <summary>
+    /// If <see cref="TreatAnyAttributePositionAsAttributeName "/> is <see langword="true"/>, returns whether the position was adjusted
+    /// </summary>
+    protected bool DidTreatAttributePositionAsAttributeName { get; private set; }
+
+    /// <summary>
+    /// If <see cref="TreatAnyAttributePositionAsAttributeName "/> is <see langword="true"/>, returns the range of the full attribute
+    /// </summary>
+    protected Range OriginalAttributeRange { get; private set; }
```

Comment:
This class is getting a little big IMO, and doing a few things, but I couldn't see a nice way to split it up, and more importantly I couldn't come up with a nice name for a class that could sit in the middle of the hierarchy and provide this functionality. `AbstractRazorDelegatingEndpointThatAdjustsCaretPositionToAttributeNamesIfPossible` is just silly :)

I don't think this is too egregious that we can't merge this, but its on a slippery slope. Open to feedback.